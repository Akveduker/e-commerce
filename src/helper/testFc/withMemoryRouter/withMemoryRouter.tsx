import { render } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom"

export const withMemoryRouter = (children: React.ReactNode) => {
    return render(
        <MemoryRouter>
            {children}
        </MemoryRouter>
    )
}