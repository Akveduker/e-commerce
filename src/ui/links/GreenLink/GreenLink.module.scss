@import "../../../styles/_vars.scss";
.link {
    font-size: inherit;
    color: $C2A;
    transition: $linkTranstion;
    text-decoration: none;
    &:hover {
        color: $C2C;
    }
    &__underline {
        text-decoration: underline;
    }
}
