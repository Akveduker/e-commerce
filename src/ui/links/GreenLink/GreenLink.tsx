import React, { FC } from 'react';
import { Link, LinkProps } from 'react-router-dom';
import s from './GreenLink.module.scss'
interface GreenLinkProps extends LinkProps {
    underline?: boolean;
}
const GreenLink: FC<GreenLinkProps> = ({ underline, children, ...props }) => {
    const className = underline ? `${s['link']} ${s['link__underline']}` : s['link']
    return (
        <Link className={className}  {...props}>
            {children}
        </Link>
    );
};

export default GreenLink;