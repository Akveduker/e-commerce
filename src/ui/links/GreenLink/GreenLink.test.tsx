import { render, screen } from "@testing-library/react"
import { withMemoryRouter } from "../../../helper/testFc/withMemoryRouter/withMemoryRouter"
import GreenLink from "./GreenLink"

describe('GreenLink', () => {
    test('Снапшот', () => {
        const { container } = withMemoryRouter(
            <GreenLink to={'/'}>
                link
            </GreenLink>
        )
        expect(screen.queryByText('link')).not.toHaveClass('link__underline')
        expect(container).toMatchSnapshot()
    })
    test('Есть в документе', () => {
        withMemoryRouter(
            <GreenLink to={'/'}>
                link
            </GreenLink>
        )
        expect(screen.queryByText('link')).toBeInTheDocument()
    })
    test('Класс меняется, снапшот', () => {
        const { container } = withMemoryRouter(
            <GreenLink to={'/'} underline>
                link
            </GreenLink>
        )
        expect(screen.queryByText('link')).toBeInTheDocument()
        expect(screen.queryByText('link')).toHaveClass('link__underline')
        expect(container).toMatchSnapshot()
    })
})