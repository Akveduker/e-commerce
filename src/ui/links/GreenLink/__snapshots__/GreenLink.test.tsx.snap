// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GreenLink Класс меняется, снапшот 1`] = `
<div>
  <a
    class="link link__underline"
    href="/"
  >
    link
  </a>
</div>
`;

exports[`GreenLink Снапшот 1`] = `
<div>
  <a
    class="link"
    href="/"
  >
    link
  </a>
</div>
`;
